### PR TITLE
Fix onAfterUpdate callback in tests

### PR DESCRIPTION
**Pull request by xintongxia**
1. onAfterUpdate should be called in every test
2. onAfterUpdate should work when data is empty

### CODE REVIEW

File: modules/test-utils/src/generate-layer-tests.js [hunk 1]
```diff
@@ -98,9 +98,10 @@ export function generateLayerTests({
   testCases.forEach(testCase => {
     testCase.title = wrapTestCaseTitle(testCase.title);
     testCase.onBeforeUpdate = onBeforeUpdate;
-    testCases.onAfterUpdate = _onAfterUpdate;
   });
 
+  testCases[2].onAfterUpdate = _onAfterUpdate;
```
Comment (xintongxia) [inline modules/test-utils/src/generate-layer-tests.js]: I am not quite sure about this, I am wondering is it necessary to call onAfterUpdate when there is no data?

Comment (Pessimistress) [reply modules/test-utils/src/generate-layer-tests.js]: There can be many test cases, you cannot only add the callback to one of them.
You can check if data is empty inside `onAfterUpdate`.

File: modules/test-utils/src/generate-layer-tests.js [hunk 2]
```diff
@@ -89,7 +89,9 @@ export function generateLayerTests({
     onAfterUpdate(params);
     // Default assert
     if (params.layer.isComposite) {
-      assert(params.subLayers.length, 'Layer should have sublayers');
+      if (Object.keys(params.layer.props).length) {
```
Comment (Pessimistress) [inline modules/test-utils/src/generate-layer-tests.js]: Why not just check if `props.data` is empty? This cannot be reliable.

Comment (xintongxia) [reply modules/test-utils/src/generate-layer-tests.js]: Because it is uncertain that no sublayer will be generated when there is no data.

Comment (Pessimistress) [reply modules/test-utils/src/generate-layer-tests.js]: What is the use case you're fixing for? Can you add description to the PR summary?

Comment (Pessimistress) [reply modules/test-utils/src/generate-layer-tests.js]: If you check `GeoJsonLayer`, it does not render any sub layers if `data` is empty. This test utility must work with the most common denominator. If a layer does not require `data` to render, you should add the additional check to its own tests.

**Pull request merged**